test/battle/weather/hail.c

```diff
@@ -53,3 +53,20 @@ SINGLE_BATTLE_TEST("Hail fails if Desolate Land or Primordial Sea are active")
         }
     }
 }
+
+DOUBLE_BATTLE_TEST("Hail deals damage based on turn order")
```

Comment:
Should we have analogous tests for Sandstorm?

Reply:
done